cider/exec/plan/substrait/SubstraitPlan.cpp

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "SubstraitPlan.h"
+
+namespace cider::plan {
+
+SubstraitPlan::SubstraitPlan(const substrait::Plan& plan) : plan_(plan) {}
+
+bool SubstraitPlan::hasAggregateRel() const {
```

Comment:
Better name ```hasStatefulOp```? More important, how to differentiate group-by and non-group-by? For non-group-by, we need to return true even it's typed as ```has_aggregate```.

Reply:
stateful is only the concept in batchProcessor, for this utitily class it would be better to follow the same name  inside substrait plan